### ISSUE TITLE
hostpath.type as `DirectoryOrCreate` does not work on windows

### ISSUE THREAD

**Issue by andyzhangx** [kubernetes/kubernetes#62121]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

> Uncomment only one, leave it on its own line: 
>
> /kind bug
> /kind feature

**What happened**:
hostpath.type as `DirectoryOrCreate` does not work on windows, there will be mount failed error

```
---
apiVersion: v1
kind: Pod
metadata:
  name: aspnet-hostpath
spec:
  containers:
  - image: microsoft/aspnet
    name: aspnet-hostpath
    volumeMounts:
    - name: test-volume
      mountPath: '/azure'
  nodeSelector:
    beta.kubernetes.io/os: windows
  volumes:
  - name: test-volume
    hostPath:
      path: 'c:\var\lib\kublet\test'
      type: DirectoryOrCreate
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.11
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/sig storage
/sig windows
/assign

**Comment by dixudx** [comment]
@andyzhangx I submitted another alternative PR to fix this issue. PTAL #62154

**Comment by andyzhangx** [comment]
This issue is fixed in v1.9.7, v1.10.2, v1.11.0
